platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/describer/ConfigurationDoesNotExistFailureDescriber.java

```diff
@@ -20,36 +20,33 @@
+import org.gradle.internal.component.resolution.failure.exception.VariantSelectionByNameException;
+import org.gradle.internal.component.resolution.failure.type.ConfigurationDoesNotExistFailure;
 import org.gradle.util.Path;
 
 import java.util.Optional;
 
 /**
- * A {@link ResolutionFailureDescriber} that describes a {@link RequestedConfigurationNotFoundFailure}.
+ * A {@link ResolutionFailureDescriber} that describes a {@link ConfigurationDoesNotExistFailure}.
  */
-public abstract class RequestedConfigurationNotFoundFailureDescriber extends AbstractResolutionFailureDescriber<RequestedConfigurationNotFoundFailure> {
+public abstract class ConfigurationDoesNotExistFailureDescriber extends AbstractResolutionFailureDescriber<ConfigurationDoesNotExistFailure> {
     @Override
-    public ConfigurationSelectionException describeFailure(RequestedConfigurationNotFoundFailure failure, Optional<AttributesSchemaInternal> schema) {
-        String message = buildConfigurationNotFoundFailureMsg(failure);
+    public VariantSelectionByNameException describeFailure(ConfigurationDoesNotExistFailure failure, Optional<AttributesSchemaInternal> schema) {
+        String message = buildFailureMsg(failure);
 
         ImmutableList.Builder<String> resolutions = ImmutableList.builder();
-        boolean isLocalComponent = failure.getRequestedComponentId() instanceof ProjectComponentIdentifier;
+        boolean isLocalComponent = failure.getTargetComponent() instanceof ProjectComponentIdentifier;
         if (isLocalComponent) {
-            ProjectComponentIdentifierInternal id = (ProjectComponentIdentifierInternal) failure.getRequestedComponentId();
+            ProjectComponentIdentifierInternal id = (ProjectComponentIdentifierInternal) failure.getTargetComponent();
             Path outgoingVariantsPath = id.getIdentityPath().append(Path.path("outgoingVariants"));
             resolutions.add("To determine which configurations are available in the target project, run " + outgoingVariantsPath.getPath());
         }
 
         resolutions.addAll(buildResolutions(suggestReviewAlgorithm()));
-        return new ConfigurationSelectionException(message, failure, resolutions.build());
+        return new VariantSelectionByNameException(message, failure, resolutions.build());
     }
 
-    private static String buildConfigurationNotFoundFailureMsg(RequestedConfigurationNotFoundFailure failure) {
-        return String.format(
-            "A dependency was declared on configuration '%s' of '%s' but no variant with that configuration name exists.",
-            failure.getRequestedName(), failure.getRequestedComponentId().getDisplayName()
-        );
+    private String buildFailureMsg(ConfigurationDoesNotExistFailure failure) {
+        return String.format("A dependency was declared on configuration '%s' which is not declared in the descriptor for '%s'.", failure.getRequestedConfigurationName(), failure.getTargetComponent().getDisplayName());
```

Comment:
We should talk about this error message.

What is a descriptor?

I believe it is a legacy term coming from [Ivy desriptor](https://ant.apache.org/ivy/history/latest-milestone/ivyfile.html) and we should not use this terminology.

Reply:
Yeah, this one is no good.  I will add this to the list of future work, but am not going to address it here.

Reply:
But in this PR you _changed_ it to this wording. 

This should be addressed here by either reverting or choosing a more appropriate wording

Reply:
So I did.  Merge/rebase mistake then.  I'll revert it.